subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/services/ConfigurationCacheEnvironmentChangeTracker.kt

```diff
@@ -102,11 +113,23 @@ class EnvironmentChangeTracker(private val problemFactory: ProblemFactory) {
         }
 
         override fun toRestoring(): Restoring {
-            return Restoring()
+            return Restoring(
+                mutatedSystemProperties
```

Comment:
Can you please add a clarifying comment here, when this happens and why it is important to check overridden properties when restoring? We spent quite a lot of time figuring this out, and I'm not sure that the comment in 196 is enough to recall all the details.